Source/gui/SNTNotificationManager.m

```diff
@@ -125,6 +127,8 @@ - (void)queueMessage:(SNTMessageWindowController *)pendingMsg {
   [self.pendingNotifications addObject:pendingMsg];
 
   if (!self.currentWindowController) {
+    // Add app to Cmd+Tab and Dock.
+    NSApp.activationPolicy = NSApplicationActivationPolicyRegular;
```

Comment:
I think `queueMessage:` is called from not-the-main-thread. Does NSApp care?

Reply:
To that point, `self.pendingNotifications` seems to be accessed and written to on different threads. We should also fix that.

Reply:
NSApp doesn't seem to care but the threading issue is a bit surprising. It looks like only queueMessage: is called from a non-main thread so I've moved much of it into a dispatch block.

Reply:
Nice.